File: api/internal/gateway/user/v1/response/promotion.go
package response

// Promotion - プロモーション情報
type Promotion struct {
	ID           string `json:"id"`           // プロモーションID
	Title        string `json:"title"`        // タイトル
	Description  string `json:"description"`  // 詳細説明
	DiscountType int32  `json:"discountType"` // 割引計算方法
	DiscountRate int64  `json:"discountRate"` // 割引額(%/円)
	Code         string `json:"code"`         // クーポンコード
	StartAt      int64  `json:"startAt"`      // クーポン使用可能日時(開始)
	EndAt        int64  `json:"endAt"`        // クーポン使用可能日時(終了)
}

File: web/admin/src/types/validations/promotion.ts
import type { ValidationArgs } from '@vuelidate/core'
import { maxLength, minLength, minValue, required } from '~/lib/validations'

export const CreatePromotionValidationRules: ValidationArgs = {
  title: { required, maxLength: maxLength(64) },
  description: { required, maxLength: maxLength(2000) },
  discountType: {},
  discountRate: { minValue: minValue(0) },
  code: { required, minLength: minLength(8), maxLength: maxLength(8) },
}

export const UpdatePromotionValidationRules: ValidationArgs = {
  title: { required, maxLength: maxLength(64) },
  description: { required, maxLength: maxLength(2000) },
  discountType: {},
  discountRate: { minValue: minValue(0) },
  code: { required, minLength: minLength(8), maxLength: maxLength(8) },
}
